tests/components/homekit/test_type_switches.py

```diff
@@ -140,32 +143,58 @@ async def test_switch_set_state(
     assert events[-1].data[ATTR_VALUE] is None
 
 
-async def test_valve_set_state(hass: HomeAssistant, hk_driver, events) -> None:
+@pytest.mark.parametrize(
+    ("domain", "accessory_class", "service_on", "service_off", "state_on", "state_off"),
+    [
+        ("switch", ValveSwitch, "turn_on", "turn_off", STATE_ON, STATE_OFF),
+        ("valve", Valve, "open_valve", "close_valve", STATE_OPEN, STATE_CLOSED),
+    ],
+)
+async def test_valve_set_state(
+    hass: HomeAssistant,
```

Comment:
Please split this into two tests as we aren't allowed to have `if` branching in the body of new tests

Reply:
I have split these tests.